attribute-projection-functions/src/main/java/org/hypertrace/core/attribute/service/projection/functions/Hash.java

```diff
@@ -0,0 +1,20 @@
+package org.hypertrace.core.attribute.service.projection.functions;
+
+import static java.util.Objects.nonNull;
+
+import com.github.f4b6a3.uuid.UuidCreator;
+import com.github.f4b6a3.uuid.creator.rfc4122.NameBasedSha1UuidCreator;
+import java.util.UUID;
+import javax.annotation.Nullable;
+
+public class Hash {
+
+  private static final NameBasedSha1UuidCreator HASHER =
+      UuidCreator.getNameBasedSha1Creator()
+          .withNamespace(UUID.fromString("5088c92d-5e9c-43f4-a35b-2589474d5642"));
```

Comment:
is this one of those magic guids that should never change?

Reply:
Yep - I planned on making this unique per tenant, but in the OLAP context we won't have that kind of information (unless provided explicitly). A UUIDv5 requires a namespace (actually, this lib seems to allow without but I don't know what the implications of that are). So this is just a randomly generated UUID - I didn't keep it in its own constant since by design it should never be referenced elsewhere.

Thoughts? Clearer as an explicit constant?

Reply:
Actually as hardcoded it's better and all consumers can call this one method. Maybe some comments around it?

Reply:
sure, will add a comment

Reply:
Is there any relation between this and the UUID generated in entity-service?

Reply:
Good catch - I completely forgot that I shared the same namespace with that UUID generator. So right now, there's no relationship, but I shared the namespace because the plan is to eventually remove that generator and have things like entity IDs purely exist as projected attributes. The exact migration there is TBD.

Reply:
Does this mean we don't need to change entity-service id generation logic to be not fully dependent on identifying properties?

Reply:
Confused by the double negative here, but the idea is that we're flipping the burden of ID generation. Right now, as I understand it, we upsert an entity by setting attribute values, on an entity object, sending it to entity service, and then reading back the ID from the response. Instead, the plan is to make this formation of an entity config driven rather than code driven - that is, rather than creating the entity explicitly by picking the identifying attributes, we'll have an attribute that is defined as the id attribute for an entity and can be read through the trace client. Entity formation/liveness can then be completely decoupled from enrichment, but use the same mechanism to get the ID. There's a number of TBDs here of course, but that's the general idea.